### DISCUSSION TITLE
Violating Nominatim Usage Policy

### DISCUSSION THREAD

**Discussion by tsziming** [tomik23/Leaflet.Autocomplete#23] [Announcements]
According to https://operations.osmfoundation.org/policies/nominatim/ :

> Unacceptable Use
The following uses are strictly forbidden and will get you banned:
> - Auto-complete search This is not yet supported by Nominatim and you must not implement such a service on the client side using the API.
...

It seems to me that your repo violates Nominatim usage policy by providing client-side autocomplete using Nominatim API. I would propose using other public APIs like https://photon.komoot.io instead in order to stick with fair use.

**Comment by tomickigrzegorz** [comment]
Nobody is forcing you to use nominatim, this is just an example of using the library. You can set up a nominatim server yourself or use some api (as it is in the description) a library and use it in your own project.

**Comment by tsziming** [comment]
Good point, so I agree that the code itself doesn't violate the fair use policy. 

However, I am still not sure about the demo (this one https://tomickigrzegorz.github.io/Leaflet.Autocomplete/). It is a public and hosted resource and it may be considered a violation. What do you think?

**Reply by tomickigrzegorz** [reply]
I still don't understand, the problem is that I'm using nominatim in the library or what? Because if so, no one is forcing you to use it.
The library was not created just for nominatim use but for files in general, data array or even geojson usage.
I recommend seeing the bottom of the page [Leaflet.Autocomplete](https://github.com/tomickigrzegorz/Leaflet.Autocomplete)

If you go to the leafeltjs website in the plugins tab, you will see a few examples of nominatim usage in the examples.

If someone uses a library, they should get acquainted with the license and what the library uses.
Okay, I'm closing this pointless discussion because it doesn't contribute anything.

I will leave a link to [nominatim policies](https://operations.osmfoundation.org/policies/nominatim/)
